doc/CrudSamples.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright © 2017 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+import com.cloudant.sync.documentstore.Attachment;
+import com.cloudant.sync.documentstore.DocumentBodyFactory;
+import com.cloudant.sync.documentstore.DocumentRevision;
+import com.cloudant.sync.documentstore.DocumentStore;
+import com.cloudant.sync.documentstore.UnsavedFileAttachment;
+import com.cloudant.sync.documentstore.UnsavedStreamAttachment;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+ * Code samples to show how to use CRUD (create, read, update, delete)
+ * features of the library
+ */
+public class CrudSamples {
+
+    public void crudSamples() throws Exception {
+        
+        // DocumentStore and Database:
+        // The examples below show how to obtain a DocumentStore instance how to obtain the Database
+        // object it owns.
+
+        // Choose a path for your DocumentStores
+        File path = new File("/tmp");
+        // on Android, we could do something like:
+        // File path = getApplicationContext().getDir("document_stores");
+
+        // Once you've got a path, it's straightforward to create DocumentStores:
+
+        DocumentStore ds = DocumentStore.getInstance(new File(path, "my_document_store"));
+        DocumentStore ds2 = DocumentStore.getInstance(new File(path, "other_document_store"));
+
+        // Static getInstance() methods on DocumentStore will create and initial a DocumentStore
+        // instance if it doesn't already exist, or retrieve an existing one.
+
+        // To close a DocumentStore (which closes the underlying database files and release native
+        // resources):
+        ds.close();
+
+        // To delete a DocumentStore (which closes it first):
+        ds.delete();
+
+        // Document CRUD APIs
+
+        // Once you have a DocumentStore instance, you can use it to create, update and
```

Comment:
what about read? admittedly if you've just created the store there won't be anything to read yet.